src/Generators/Base.php

```diff
@@ -117,7 +117,10 @@ public function getOptions()
      */
     protected function getGenerator()
     {
+        if ($this->kind === false) return $this->kind;
```

Comment:
Okay, where should this go, then? I think enforcing string here is either incorrect, or this check needs to be done outside this method. What do you suggest?

Reply:
Where is it documented that this could be set to false? The phpdoc for the kind and the constructor say only strings are allowed.

Reply:
As highlighted in #456, if I set an attribute's value to false as part of my test, it returns an empty string ' '. This breaks queries that have strict mode enabled, not to mention it's using a value that is unexpected.

Ignore my comment above - what would you suggest I do instead? I can update the Generic generator that checks for the value there, and if false simply returns. That's the only other location that it makes sense, I believe.

Reply:
Note that this also happens if you use a boolean generator and the generator returns false (instead of true), with strict mode enabled.

Reply:
My tests also pass (both on local project and FactoryMuffin), when moved to the Generic generate method.

Reply:
I think the correct usage would be to use the "callable generator" (in 3.0, in 2.1 is specific to only closures), and set the attribute to `function () { return false; }`.

Out of interest, why are you stuck on 2.1, rather than 3.0?

Reply:
> Note that this also happens if you use a boolean generator and the generator returns false (instead of true), with strict mode enabled.

Oh. Hmm. I think I need to pull down the whole codebase and have a proper read over of the code again. Are you actually using 2.1? I'd rather only fix this in 3.x.

Reply:
Turns out upgrading to 3.0 is going to be an immense amount of work due to the move away from the facade and the faker changes. We may come back to this but for now I don't have the days it's going to take me to upgrade our test suite (we have 2300 tests, not including acceptance tests).